crates/accelerate/src/synthesis/qft/qft_decompose_lnn.rs

```diff
@@ -0,0 +1,112 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2023.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::synthesis::linear_phase::cz_depth_lnn::LnnGatesVec;
+use crate::synthesis::permutation::_append_reverse_permutation_lnn_kms;
+use pyo3::prelude::*;
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::operations::{Param, StandardGate};
+use qiskit_circuit::Qubit;
+use smallvec::smallvec;
+use std::f64::consts::PI;
+
+/// Construct a circuit for the Quantum Fourier Transform using linear
+/// neighbor connectivity.
+///
+/// The construction is based on Fig 2.b in Fowler et al. [1].
+///
+/// .. note::
+///
+///     With the default value of ``do_swaps = True``, this synthesis algorithm creates a
+///     circuit that faithfully implements the QFT operation. When ``do_swaps = False``,
+///     this synthesis algorithm creates a circuit that corresponds to "QFT-with-reversal":
+///     applying the QFT and reversing the order of its output qubits.
+///
+/// Args:
+///     num_qubits: The number of qubits on which the Quantum Fourier Transform acts.
+///     approximation_degree: The degree of approximation (0 for no approximation).
+///         It is possible to implement the QFT approximately by ignoring
+///         controlled-phase rotations with the angle beneath a threshold. This is discussed
+///         in more detail in https://arxiv.org/abs/quant-ph/9601018 or
+///         https://arxiv.org/abs/quant-ph/0403071.
+///     do_swaps: Whether to synthesize the "QFT" or the "QFT-with-reversal" operation.
+///
+/// Returns:
+///     A circuit implementing the QFT operation.
+///
+/// References:
+///     1. A. G. Fowler, S. J. Devitt, and L. C. L. Hollenberg,
+///        *Implementation of Shor's algorithm on a linear nearest neighbour qubit array*,
+///        Quantum Info. Comput. 4, 4 (July 2004), 237–251.
+///        `arXiv:quant-ph/0402196 [quant-ph] <https://arxiv.org/abs/quant-ph/0402196>`_
+
+#[pyfunction]
+#[pyo3(signature=(num_qubits, do_swaps=true, approximation_degree=0))]
+pub(super) fn _synth_qft_line(
```

Comment:
The leading underscore seems more like a Python thing, I don't think we need this custom here and rely on the `pub` qualifier to control function privacy
```suggestion
pub(super) fn synth_qft_line(
```